src/solvers/dgsem_p4est/dg_2d.jl

```diff
@@ -364,11 +363,8 @@ end
 
     # Call pointwise numerical flux function for the conservative part
     # in the normal direction on the boundary
-    flux_ = boundary_condition(u_inner, normal_direction, x, t, surface_flux, equations)
-
-    # Compute pointwise nonconservative numerical flux at the boundary.
-    noncons_ = boundary_condition(u_inner, normal_direction, x, t, nonconservative_flux,
-                                  equations)
+    flux_, noncons_ = boundary_condition(u_inner, normal_direction, x, t,
```

Comment:
Maybe for consistency with the `TreeMesh` implementations?

```suggestion
    flux, noncons_flux = boundary_condition(u_inner, normal_direction, x, t,
```

In that case, below becomes (can't suggest it here)

```julia
surface_flux_values[v, node_index, direction_index, element_index] = flux[v] + 0.5f0 * noncons_flux[v]
```

Reply:
Thanks! I agree with all your suggestions!